src/UI/Index/Main.hs

```diff
@@ -126,4 +125,9 @@ renderAuthors isSelected authors =
 renderTagsWidget :: [Tag] -> Tag -> Widget Name
 renderTagsWidget tgs ignored =
     let ts = filter (/= ignored) tgs
-    in withAttr mailTagsAttr $ vLimit 1 $ txt $ decodeLenient $ B.intercalate " " $ fmap getTag ts
+        render tag = padRight (Pad 1) $ withAttr (toAttrName tag) $ txt (decodeLenient $ getTag tag)
+    in vLimit 1 $ hBox $ render  <$> ts
+
+toAttrName :: Tag -> AttrName
+toAttrName t = let attr = attrName . unpack . decodeLenient . getTag
+               in attr t
```

Comment:
We can clean this up:

```haskell
toAttrName = attrName . unpack . decodeLenient . getTag
```

And, to elaborate what I suggested above, this would become like:

```haskell
toAttrName = ("tag" <>) . attrName . unpack . decodeLenient . getTag
```

Reply:
AAah yes I see what you mean. Will try.

Reply:
Hm... yeh.. your idea of returning the `"tag"` default style won't work with our current setup. The function `toAttrName` will always return a tag which means you always return `"tag"` with `"something"`.

If you wanted to support a fallback just for `"tag"`, we'd need a separate `AttrMap` and set the default attribute of that map accordingly. At the moment it's basically a global map. At this point, if you want colored tags, just use the what's in the default config and you're good to go.

Having said that, I'd still keep your suggestions.

Reply:
Right, but can't we change the config to refer to `"tag" <> "foo"`, and it will work as expected?

Reply:
I haven't tried anything with attrmaps yet,  so maybe my intuition / interpretation of the docs is wrong...

Reply:
Hm not sure what you mean by

> change the config to refer to `"tag" <> "foo"`

The current map is global and stores attributes for the entire application. If you lookup an attribute which isn't in the map, it's rendered with the default attribute, currently `defAttr` from `Vty`. That happens whether we use it for tags or whatever. If we change the `defAttr` to be the color definition for a default Tag than any other undefined element would use the same color definition as the default tag.

The other alternative I could imagine is to query the `AttrMap` if `"tag" <> "foo"` is in it and if not return the default...

Reply:
AttrMap lookup is heirarchical.  i.e. if you look for `"foo"<>"bar"` in a map that only has `"foo"`, then `"foo"` style will apply.

So you can have a style for `"tag"`, which is the default for all attrs beginning with `"tag"`, and then override it for more specific tags (e.g. `"tag"<>"inbox"`).

At least, that is my interpretation of the docs: https://hackage.haskell.org/package/brick-0.47/docs/Brick-AttrMap.html#v:attrMapLookup